agent/src/beerocks/slave/ap_manager_thread.cpp

```diff
@@ -1527,7 +1528,7 @@ void ap_manager_thread::handle_hostapd_attached()
             usleep(ACS_READ_SLEEP_USC);
         }
     } else {
-        ap_wlan_hal->read_supported_channels();
+        ap_wlan_hal->update_preference_channels_from_supported_channels();
```

Comment:
No - the logic of how we choose the channel preference should be abstracted by the BWL.
So instead we should have a `read_preferred_channels()` API in BWL which **internally** will use the radio supported channels **we read at boot** if ACS is disabled.
So - this whole commit should be removed, and replaced with a simple clear API to read preferred channels exactly like the API to read supported channels.